lib/model/QubitDigitalObject.php

```diff
@@ -2562,21 +2562,31 @@ public static function convertAudioToMp3($originalPath, $newPath)
             return false;
         }
 
-        $command = 'ffmpeg -y -i '.$originalPath.' '.$newPath.' 2>&1';
-        exec($command, $output, $status);
+        $format = self::readStreamInformation($originalPath);
+        
+        error_log(json_encode($format, JSON_PRETTY_PRINT));
 
-        if ($status) {
-            $error = true;
+        $formatName = $format['format_name'] ?? null;
 
-            for ($i = count($output) - 1; $i >= 0; --$i) {
-                if (strpos($output[$i], 'output buffer too small')) {
-                    $error = false;
+        $reformatFile = $formatName !== 'mp3' ||
+                        strtolower(pathinfo($originalPath, PATHINFO_EXTENSION)) !== 'mp3';
+        
+        error_log("Format name: ".$formatName);
 
-                    break;
-                }
-            }
+        $status = null;
+        if ($reformatFile) {
+            error_log("Reformatting audio file to mp3");
+            $command = sprintf('ffmpeg -y -i %s %s 2>&1', escapeshellarg($originalPath), escapeshellarg($newPath));
+            exec($command, $output, $status);
+        } else {
+            error_log("Copying audio file to new path, no reformatting required");
+            $status = copy($originalPath, $newPath) ? 0 : 1;
         }
 
+        if ($status !== 0) {
+            return false;
+        }  
```

Comment:
I'm worried that they were doing some checking in that bit of code you removed for whether a non-zero exit code was actually a non-issue.

This might cover our bases better:

```suggestion
        if ($status !== 0 && !file_exists($newPath)) {
            return false;
        }  
```

Reply:
This looks good to me.